### PR TITLE
Add IsQueuedForDeletion

### PR DESCRIPTION
**Pull request by ElectroJr**
Allows systems to check if an entity is about to be deleted before doing potentially unnecessary & expensive computations (e.g., ExtensionCableSystem)

### CODE REVIEW

File: Robust.Shared/GameObjects/EntityManager.cs [hunk 1]
```diff
@@ -34,8 +34,7 @@ public partial class EntityManager : IEntityManager
         /// <inheritdoc />
         public virtual IEntityNetworkManager? EntityNetManager => null;
 
-        protected readonly Queue<EntityUid> QueuedDeletions = new();
```
Comment (gradientvera) [inline Robust.Shared/GameObjects/EntityManager.cs]: The reason we kept a queue as well was to maintain deletion order...

**Review by gradientvera — CHANGES_REQUESTED** [review]

File: Robust.Shared/GameObjects/EntityManager.cs (hunk 1 above)
Comment (ElectroJr) [reply Robust.Shared/GameObjects/EntityManager.cs]: ok, reverted that change

File: Robust.Shared/GameObjects/EntityManager.cs [hunk 2]
```diff
@@ -297,10 +297,12 @@ private void RecursiveDeleteEntity(EntityUid uid)
 
         public void QueueDeleteEntity(EntityUid uid)
         {
-            if(QueuedDeletionsSet.Add(uid))
+            if (QueuedDeletionsSet.Add(uid))
                 QueuedDeletions.Enqueue(uid);
         }
 
+        public bool IsQueuedForDeletion(EntityUid uid) => QueuedDeletions.Contains(uid);
```
Comment (ShadowCommander) [inline Robust.Shared/GameObjects/EntityManager.cs]: Is it faster to check the `QueuedDeletionsSet`?

Comment (ElectroJr) [reply Robust.Shared/GameObjects/EntityManager.cs]: AFAIK checking the hashset is faster, this was an accidental leftover from before where I had removed the queue and just had the hashset. I'll update this now.

**Review by ShadowCommander — APPROVED** [review]

**Review by metalgearsloth — APPROVED** [review]
conflicts

**Pull request merged**